hikari/impl/cache.py

```diff
@@ -1485,7 +1482,7 @@ def _garbage_collect_message(
         if message.object.id in self._referenced_messages:
```

Comment:
This should return None/False/whatever you feel like if it was already marked as deleted

Reply:
The uses for this function stay the same, so either its a bug I didnt fix or I did in fact miss something. I dont get the point for an extra return tho

Reply:
Clear and delete methods should only be returning objects which were actually marked as deleted not objects which are being kept alive by references